src/org/opensolaris/opengrok/configuration/messages/Message.java

```diff
@@ -84,6 +83,19 @@
     public abstract byte[] apply(RuntimeEnvironment env) throws Exception;
 
     /**
+     * Validate the current message and throw an exception if the message is not
+     * valid. Further implementation can override this method to get the message
+     * into a state they would expect.
+     *
+     * @throws Exception
+     */
+    public void validate() throws Exception {
```

Comment:
Class name has only meaning for normal message so I don't validate it. That's a bad original design for now.